### PR TITLE
feat: add some pre-commit hooks

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -18,63 +18,65 @@ classifiers =
     Programming Language :: Python :: 3.7
     Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
```
Comment (parano) [inline setup.cfg:21]: do we have CI enabled for 3.10 already?

File: pyproject.toml [hunk 2]
```diff
@@ -108,7 +108,11 @@ min-public-methods=2
 
 [tool.isort]
 profile = "black"
-skip_glob = ["typings"]
+length_sort = true
```
Comment (bojiang) [inline pyproject.toml:111]: Is there any conflict between the isort and formater options?

File: hooks/pre-commit [hunk 3]
```diff
@@ -8,7 +8,22 @@ VALID_BRANCH_REGEX="^(style|chore|refactor|feat|internal|bugfix|improvement|libr
...
+# TODO: enable black
+# black --config ./pyproject.toml bentoml/ tests/ docker/
+
+# black --config ./pyproject.toml --pyi typings/**/*.pyi
+
+# TODO: enable isort
+# isort .
+
+for f in $(find .github/ -type f -name '*.yml'); do
+  yamllint -c "$GIT_ROOT"/.yamllint.yml "$f" || exit 1
```
Comment (bojiang) [inline hooks/pre-commit:25]: putting this into pre-commit will force contributors having `yamllint` installed right? How about windows?

Comment (parano) [reply hooks/pre-commit:25]: yamllint is just a python package right?

File: pyproject.toml (hunk 2 above)
Comment (aarnphm) [reply pyproject.toml:111]: like black and isort?

File: hooks/pre-commit (hunk 3 above)
Comment (aarnphm) [reply hooks/pre-commit:25]: it is just a pip package

File: pyproject.toml (hunk 2 above)
Comment (bojiang) [reply pyproject.toml:111]: yep.

Comment (bojiang) [reply pyproject.toml:111]:
```
[PASS] black format check passed!
11
Running isort format check...
12
ERROR: /home/runner/work/BentoML/BentoML/tests/integration/frameworks/test_pytorch_lightning_impl.py Imports are incorrectly sorted and/or formatted.
13
ERROR: /home/runner/work/BentoML/BentoML/tests/integration/frameworks/test_onnx_impl.py Imports are incorrectly sorted and/or formatted.
14
ERROR: /home/runner/work/BentoML/BentoML/tests/integration/frameworks/test_catboost_impl.py Imports are incorrectly sorted and/or formatted.
15
ERROR: /home/runner/work/BentoML/BentoML/tests/integration/frameworks/test_keras_impl.py Imports are incorrectly sorted and/or formatted.
16
ERROR: /home/runner/work/BentoML/BentoML/tests/integration/frameworks/test_lightgbm_impl.py Imports are incorrectly sorted and/or formatted.
```

Comment (bojiang) [reply pyproject.toml:111]: just tested. They're compatible with each other

**Review by bojiang — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (aarnphm) [reply pyproject.toml:111]: ok

**Comment by aarnphm** [comment]
I will merge this in so that the runner app PR won't get blocked

File: setup.cfg (hunk 1 above)
Comment (parano) [reply setup.cfg:21]: let's remove it then

**Pull request merged**